oasislmf/preparation/gul_inputs.py

```diff
@@ -44,12 +45,61 @@
 pd.options.mode.chained_assignment = None
 warnings.simplefilter(action='ignore', category=FutureWarning)
 
+VALID_OASIS_GROUP_COLS = [
+        'item_id',
+        'peril_id',
+        'coverage_id',
+        'coverage_type_id',
+        'peril_correlation_group'
+    ]
+
+
+def process_group_id_cols(group_id_cols: List[str], exposure_df_columns: List[str], correlations: bool) -> List[str]:
+    """
+    cleans out columns that are not valid oasis group columns.
+
+    Valid group id columns can be either
+    1. exist in the location file
+    2. be listed as a useful internal col
+
+    Args:
+        group_id_cols: (List[str]) the ID columns that are going to be filtered
+        exposure_df_columns: (List[str]) the columns in the exposure dataframe
+        correlations: (bool) if set to True means that we are hashing with correlations in mind therefore the
+                             "peril_correlation_group" column is added
+
+    Returns: (List[str]) the filtered columns
+    """
+    for col in VALID_OASIS_GROUP_COLS:
+        if col not in list(exposure_df_columns) + VALID_OASIS_GROUP_COLS:
```

Comment:
this is merely legacy code that has been moved to another area but happy to chance this:
https://github.com/OasisLMF/OasisLMF/blob/1c59d93b589d9c119178a914900ef2e21f43917f/oasislmf/preparation/gul_inputs.py#L156

Reply:
@sstruzik is right here, there is an issue with the logic, and will always evaluate to `False`

its checking for valid group_id columns by looking over the list `VALID_OASIS_GROUP_COLS` instead of checking the input given to the function from `group_id_cols`

Reply:
https://github.com/OasisLMF/OasisLMF/pull/1069/commits/5f6933adc8ada030be82f15f7cf969d79a12beac